src/ResourceManager/Resources/Commands.ResourceManager/Cmdlets/Handlers/RetryHandler.cs

```diff
@@ -56,15 +56,15 @@ public class RetryHandler : DelegatingHandler
         protected override async Task<HttpResponseMessage> SendAsync(HttpRequestMessage request, CancellationToken cancellationToken)
         {
             HttpResponseMessage response = null;
+
             for (int attempt = 0; attempt < RetryHandler.MaxAttempts; ++attempt)
             {
                 try
                 {
-                    response = await base
-                        .SendAsync(request: request, cancellationToken: cancellationToken)
+                    response = await base.SendAsync(request: request, cancellationToken: cancellationToken)
                         .ConfigureAwait(continueOnCapturedContext: false);
-
-                    if (attempt == RetryHandler.MaxAttempts ||
+                    
+                    if (attempt == RetryHandler.MaxAttempts - 1 ||
```

Comment:
Why do we need the attempt == RetryHandler.MaxAttempts -1 condition here? The loop condition will ensure attempt won't go above MaxAttempts (3).